### ISSUE TITLE
VisualBert: Why isn't the pooler_output used to calculate the logits in VisualBertForQuestionAnswering?

### ISSUE THREAD

**Issue by niktheod** [huggingface/transformers#35025]
I noticed that in the source code of `VisualBertForQuestionAnswering`, the embedding vector that is used to calculate the logits is not the `pooler_output` that is returned from `VisualBertModel`, but rather the embedding representation of the last text token in `last_hidden_state` (also returned from `VisualBertModel`), as you can see in the [code](https://github.com/huggingface/transformers/blob/052e652d6d53c2b26ffde87e039b723949a53493/src/transformers/models/visual_bert/modeling_visual_bert.py#L1242) below:

        sequence_output = outputs[0]

        # TO-CHECK: From the original code
        index_to_gather = (
            index_to_gather.unsqueeze(-1).unsqueeze(-1).expand(index_to_gather.size(0), 1, sequence_output.size(-1))
        )
        pooled_output = torch.gather(sequence_output, 1, index_to_gather)

Why is that? Usually in similar models the `pooler_output` is used to calculate the logits, which comes from the [CLS] token in the beginning of the text.

**Comment by Rocketknight1** [comment]
Since the comment indicates that it's from the original code, my guess is that's what the original authors used! Both the final token and the pooler are viable choices here.
